File: tests/fixtures/edgecase_repo/kotlin/Service1.kt
package com.example.gen

import com.example.gen.Base1

interface Handler1 {
    fun handle(payload: String): Boolean
}

open class Base1(val name: String)

class Service1(name: String) : Base1(name), Handler1 {
    override fun handle(payload: String): Boolean {
        validate(payload)
        return true
    }

    private fun validate(payload: String) {}
}

fun helper1(value: String): String {
    return value
}

File: tests/fixtures/edgecase_repo/swift/Service1.swift
import Foundation

protocol Handler1 {
    func handle(_ payload: String) -> Bool
}

class Base1 {
    let name: String
    init(name: String) {
        self.name = name
    }
}

class Service1: Base1, Handler1 {
    func handle(_ payload: String) -> Bool {
        self.validate(payload)
        return true
    }

    private func validate(_ payload: String) {}
}

func helper1(_ value: String) -> String {
    return value
}
